### ISSUE TITLE
Assign worker category spec per task

### ISSUE THREAD

**Issue by jakubmatyszewski** [apache/druid#14809]
### Description

I've recently started using MiddleManager categories and Overlords `WorkerCategorySpec` for [worker select strategy](https://druid.apache.org/docs/latest/configuration/#worker-select-strategy). As I understand currently it only supports picking MM category based on **task type**. I wonder if it's possible or it would be feasible feature to request, so that category spec would be assigned **per task**?

### Motivation

I was thinking on separating big batch ingestions from other types of ingestions, so that I could have dedicated Middlemanager instances processing such workloads. Currently I can only select by task type, but even though separating kafka ingestions from batch ingestions gets me a step closer to what I'm aiming for, I wish I could separate small batch ingestions from batch ingestions we run monthly that are exponentially bigger, so this tasks would be processed by scaled Middlemanagers dedicated only for that purpose.

**Comment by cryptoe** [comment]
+1
seems useful.

**Comment by a2l007** [comment]
@jakubmatyszewski Are you planning to proceed with this feature contribution? I'm planning to implement something similar where the select strategy could be assigned at a datasource level, which might work for your usecase as well.

**Comment by jakubmatyszewski** [comment]
@a2l007, no, we settled with some workaround and I think I won't be able to come back to this idea any time soon, but your plan sounds in fact like something that would solve some of my problems. Thanks for the mention
